### PR TITLE
Added docs for TypedNavigationInterface

### PR DESCRIPTION
**Pull request by f7deleon**
Added docs for TypedNavigationInterface

### CODE REVIEW

File: typednavigation/src/main/java/com/xmartlabs/typednavigation/TypedNavigation.kt [hunk 1]
```diff
@@ -4,7 +4,14 @@ import androidx.navigation.NamedNavArgument
 import androidx.navigation.NavType
 import androidx.navigation.navArgument
 
+/** Sealed class that implements [TypedNavigationInterface]
+ *
+ *
+ */
```
Comment (mirland) [inline typednavigation/src/main/java/com/xmartlabs/typednavigation/TypedNavigation.kt:10]:
```suggestion
/** Sealed class that implements [TypedNavigationInterface] */
```

Comment (mirland) [reply typednavigation/src/main/java/com/xmartlabs/typednavigation/TypedNavigation.kt:10]: same for other sl comments

File: typednavigation/src/main/java/com/xmartlabs/typednavigation/TypedNavigationInterface.kt [hunk 2]
```diff
@@ -3,54 +3,137 @@ package com.xmartlabs.typednavigation
 import androidx.navigation.NamedNavArgument
 import androidx.navigation.NavType
 
-
+/** An interface that implements the basic attributes needed for navigation
```
Comment (mirland) [inline typednavigation/src/main/java/com/xmartlabs/typednavigation/TypedNavigationInterface.kt:6]: don't you need a nl?
```suggestion
/**
An interface that implements the basic attributes needed for navigation
```

File: typednavigation/src/main/java/com/xmartlabs/typednavigation/TypedNavigationInterface.kt [hunk 3]
```diff
@@ -3,54 +3,137 @@ package com.xmartlabs.typednavigation
...
+ *
+ */
 sealed interface TypedNavigationInterface {
     val name: String
     val url: String
     val navArguments: List<NamedNavArgument>
 }
 
-interface TypedNavigationInterface0 : TypedNavigationInterface{
+/** An interface that implements navigation without parameters
```
Comment (mirland) [inline typednavigation/src/main/java/com/xmartlabs/typednavigation/TypedNavigationInterface.kt:19]: I don't like the implements...

maybe something like:
```suggestion
/** Defines the navigation without parameters
```

File: typednavigation/src/main/java/com/xmartlabs/typednavigation/TypedNavigationInterface.kt [hunk 4]
```diff
@@ -3,54 +3,137 @@ package com.xmartlabs.typednavigation
...
+ *
+ */
+interface TypedNavigationInterface0 : TypedNavigationInterface {
 
-    fun route() : String
+    /** Get the [String] route for the destination
+     * @return the [String] containing the route of the destination
+     */
```
Comment (mirland) [inline typednavigation/src/main/java/com/xmartlabs/typednavigation/TypedNavigationInterface.kt:26]: I'd use just the shorter way, but maybe it's not needed
```suggestion
    /** Returns the route of the destination */
```

**Review by mirland — APPROVED** [review]

File: typednavigation/src/main/java/com/xmartlabs/typednavigation/TypedNavigation.kt [hunk 5]
```diff
@@ -4,7 +4,14 @@ import androidx.navigation.NamedNavArgument
...
 sealed class TypedNavigation {
+    /** Implementation for the [TypedNavigationInterface0]
+     *
+     */
```
Comment (f7deleon) [inline typednavigation/src/main/java/com/xmartlabs/typednavigation/TypedNavigation.kt:14]:
```suggestion
    /** Implementation for the [TypedNavigationInterface0] */
```

File: typednavigation/src/main/java/com/xmartlabs/typednavigation/TypedNavigation.kt [hunk 6]
```diff
@@ -13,6 +20,9 @@ sealed class TypedNavigation {
         override fun route(): String = "$name/"
     }
 
+    /** Implementation for the [TypedNavigationInterface1]
+     *
+     */
```
Comment (f7deleon) [inline typednavigation/src/main/java/com/xmartlabs/typednavigation/TypedNavigation.kt:25]:
```suggestion
    /** Implementation for the [TypedNavigationInterface1] */
```

File: typednavigation/src/main/java/com/xmartlabs/typednavigation/TypedNavigation.kt [hunk 7]
```diff
@@ -24,6 +34,9 @@ sealed class TypedNavigation {
         override fun route(arg1: A): String = "$name/${arg1.toString()}"
     }
 
+    /** Implementation for the [TypedNavigationInterface2]
+     *
+     */
```
Comment (f7deleon) [inline typednavigation/src/main/java/com/xmartlabs/typednavigation/TypedNavigation.kt:39]:
```suggestion
    /** Implementation for the [TypedNavigationInterface2] */
```

File: typednavigation/src/main/java/com/xmartlabs/typednavigation/TypedNavigation.kt [hunk 8]
```diff
@@ -38,6 +51,9 @@ sealed class TypedNavigation {
             "$name/${arg1.toString()}/${arg2.toString()}"
     }
 
+    /** Implementation for the [TypedNavigationInterface3]
+     *
+     */
```
Comment (f7deleon) [inline typednavigation/src/main/java/com/xmartlabs/typednavigation/TypedNavigation.kt:56]:
```suggestion
    /** Implementation for the [TypedNavigationInterface3] */
```

File: typednavigation/src/main/java/com/xmartlabs/typednavigation/TypedNavigation.kt [hunk 9]
```diff
@@ -54,6 +70,9 @@ sealed class TypedNavigation {
             "$name/${arg1.toString()}/${arg2.toString()}/${arg3.toString()}"
     }
 
+    /** Implementation for the [TypedNavigationInterface4]
+     *
+     */
```
Comment (f7deleon) [inline typednavigation/src/main/java/com/xmartlabs/typednavigation/TypedNavigation.kt:75]:
```suggestion
    /** Implementation for the [TypedNavigationInterface4] */
```

File: typednavigation/src/main/java/com/xmartlabs/typednavigation/TypedNavigation.kt [hunk 10]
```diff
@@ -72,6 +91,9 @@ sealed class TypedNavigation {
             "$name/${arg1.toString()}/${arg2.toString()}/${arg3.toString()}/${arg4.toString()}"
     }
 
+    /**Implementation for the [TypedNavigationInterface5]
+     *
+     */
```
Comment (f7deleon) [inline typednavigation/src/main/java/com/xmartlabs/typednavigation/TypedNavigation.kt:96]:
```suggestion
    /**Implementation for the [TypedNavigationInterface5] */
```

**Pull request merged**